.github/workflows/sql-test-and-build-workflow.yml

```diff
@@ -19,14 +19,25 @@ on:
 
 jobs:
   build:
+    env:
+      BUILD_ARGS: ${{ matrix.os_build_args }}
     strategy:
+      # Run all jobs
+      fail-fast: false
       matrix:
-        java:
-          - 11
-          - 17
-    runs-on: ubuntu-latest
+        java: [11, 17]
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        include:
+          - os: windows-latest
+            os_build_args: -x doctest -x integTest -x jacocoTestReport -x compileJdbc
+          - os: macos-latest
+            os_build_args: -x doctest -x integTest -x jacocoTestReport -x compileJdbc
+    runs-on: ${{ matrix.os }}
 
     steps:
+    - name: Change line endings for windows
+      if: ${{ matrix.os == 'windows-latest' }}
+      run: git config --global core.autocrlf input
```

Comment:
is it required to convert crlf when git checks out files on windows? ideally developers shouldn't have to modify this for build to work

Reply:
@joshuali925 done by modifying the file similar to other files here: 

[a2b473b](https://github.com/opensearch-project/sql/pull/907/commits/a2b473b27d7d69ee3c687f8bcc90b15412f16a1b)
Then I removed this git config in the CI